sisyphus/localengine.py

```diff
@@ -282,23 +307,19 @@ def try_to_recover_task(self, task, task_id):
             rqmt = d['requested_resources']
             logpath = os.path.relpath(task.path(gs.JOB_LOG_ENGINE))
             call_with_id = task.get_worker_call(task_id)
-            call_with_id += ['--redirect_output']
             name = task.task_name()
             task_name = task.name()
             task_instance = TaskQueueInstance(call_with_id, logpath, rqmt, name, task_name, task_id)
 
-            if call_with_id != process.cmdline()[1:]:
+            if call_with_id[1:] != process.cmdline()[1:]:
                 logging.debug('Job changed, ignore this job: %i %s %s' % (pid, process.cmdline(), task_instance.call))
                 return False
 
-            if os.path.abspath(os.getcwd()) != process.cwd():
-                logging.debut('Job changed, ignore this job: %i %s %s' % (pid, os.getcwd(), process.cwd()))
-                return False
-
```

Comment:
not sure how this ever worked. 
 * `os.path.abspath(os.getcwd())` gives the path where I execute the manager (e.g. `~/setups/mysetup`) 
 * `process.cwd()` is the work folder of a specific job (e.g. `~/setups/mysetup/work/package/MyJob.somehash/work`)

Reply:
This line probably worked due to the redirect_output flag. I think a side effect of calling redirect_output is that it creates a subprocess to actually run the worker. Only the subprocess changes it's work directory, so calling cwd on the wrapper process should return the experiment directory.
I have to admit that I didn't use this property intentionally :)

Reply:
Yeah, that was probably wrong...